.github/workflows/update-os-snapshots.yml

```diff
@@ -0,0 +1,250 @@
+name: Refresh E2E Snapshots
```

Comment:
Pretty much a copy of .github/workflows/build.yml

There's a bigger effort to DRY the workflow one day ™️ 
Out of scope for now. I did not put TODO either because it's best to study all the workflows together before slating work